earthaccess/auth.py

```diff
@@ -317,60 +266,25 @@ def _get_credentials(
             self.username = username
             self.password = password
 
-            self.tokens = token_resp.json()
+            self.token = token_resp.json()
+            logger.debug(
+                f"Using token with expiration date: {self.token['expiration_date']}"
+            )
```

Comment:
This will fix the type-checking error:

```suggestion
            token = token_resp.json()
            logger.debug(
                f"Using token with expiration date: {token['expiration_date']}"
            )
            self.token = token
            self.authenticated = True
```